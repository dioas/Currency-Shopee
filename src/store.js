import Vue from 'vue'
import Vuex from 'vuex'
// var _ = require('lodash')
const axios = require('axios')
Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    totalCurrency: 0,
    dataCurrency: {},
    getListCurrency: [],
    parentValue: 100
  },
  mutations: {
    setCurrency (state, Data) {
      this.state.dataCurrency = Data
    },
    pushDataCurrency (state, Data) {
      this.state.getListCurrency.push(Data)
    },
    setParent (state, Data) {
      this.state.parentValue = Data
    },
    deleteData (state, Data) {
      this.state.getListCurrency.splice(Data, 1)
    }
  },
  actions: {
    async getAllCurrency ({ commit }) {
      axios.get('https://api.exchangeratesapi.io/latest')
        .then(function (response) {
          // handle success
          commit('setCurrency', response.data.rates)
        })
        .catch(function (error) {
          // handle error
          console.log(error)
        })
        .then(function () {
          // always executed
        })
    },
    async getNewArray ({ commit }, data) {
      commit('pushDataCurrency', data)
    },
    async setParentNumber ({ commit }, data) {
      commit('setParent', data)
    },
    async removeIndex ({ commit }, data) {
      commit('deleteData', data)
    }
  }
})
