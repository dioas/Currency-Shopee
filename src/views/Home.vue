
<template>
  <v-container grid-list-md>
    <v-layout row wrap class="wrapper-body">
      <v-layout row wrap class="head-currency">
        <v-flex md12 xs12>
          USD - United State Dollars
        </v-flex>
        <v-flex md6 xs6>
          <b>USD</b>
        </v-flex>
        <v-flex md6 xs6 text-xs-right>
          <input class="border-input" type="number" v-model="parentData" v-on:keyup="makeCurrency" >
        </v-flex>
      </v-layout>
      <v-divider ></v-divider>
      <v-flex md12 xs12>
        <v-content style="text-align:center;">
          <CardInfoCurrency/>
          <AddButton/>
        </v-content>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
import AddButton from '@/components/AddButton'
import CardInfoCurrency from '@/components/CardInfoCurrency'
import { mapActions, mapState } from 'vuex'
export default {
  name: 'home',
  props: {
    msg: String
  },
  data: () => {
    return {
      parentData: 100
    }
  },
  components: {
    AddButton, CardInfoCurrency
  },
  computed: {
    ...mapState([
      'dataCurrency'
    ])
  },
  mounted () {
    this.getAllCurrency()
  },
  methods: {
    ...mapActions([
      'getAllCurrency',
      'setParentNumber'
    ]),
    makeCurrency () {
      this.setParentNumber(this.parentData)
    }
  }
}
</script>

<style lang="less" scope>
.wrapper-body {
  border:solid 1px #ddd;
  .head-currency {
    width: 100%;
    border-bottom:solid 1px #ddd;
    padding: 15px !important;
    margin: 0 !important;
  }
}
.border-input {
  border:solid 1px #ddd;
  padding: 10px;
}
</style>
