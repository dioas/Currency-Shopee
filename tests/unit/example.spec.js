import { mount } from '@vue/test-utils'
import Home from '@/App.vue'
import Main from '@/components/AddButton'
import store from '@/store.js'

test('renders correctly', () => {
  const wrapper = mount(Home)
  expect(wrapper.element).toMatchSnapshot()
})
describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Home)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
test('renders correctly', () => {
  const wrapper = mount(Main)
  expect(wrapper.element).toMatchSnapshot()
})
describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Main)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
const wrapper = mount(Main)
wrapper.find('button').trigger('click')

test('renders correctly', () => {
  const wrapper = mount(store)
  expect(wrapper.element).toMatchSnapshot()
})
describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(store)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})

test('renders correctly', () => {
  const wrapper = mount(Main)
  expect(wrapper.element).toMatchSnapshot()
})
describe('Component', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Main)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
