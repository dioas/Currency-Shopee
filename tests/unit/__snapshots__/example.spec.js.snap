// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="app"
>
  <div
    id="nav"
  >
    
    Currency World
  
  </div>
   
  <router-view />
</div>
`;

exports[`renders correctly 2`] = `
<div>
  <button
    class="white--text v-btn theme--light blue-grey"
    type="button"
  >
    <div
      class="v-btn__content"
    >
      <i
        aria-hidden="true"
        class="v-icon v-icon--right material-icons theme--dark"
      >
        add
      </i>
      
    Add More Currency
  
    </div>
  </button>
   
  <div
    style="display: none;"
  >
    <input
      class="input-isi"
      placeholder="Input Currency"
    />
     
    <button
      class="white--text v-btn theme--light blue-grey"
      type="button"
    >
      <div
        class="v-btn__content"
      >
        
      Submit
      
      </div>
    </button>
  </div>
</div>
`;

exports[`renders correctly 3`] = `
<!--function (el) {
      var obj;

      var args = [], len = arguments.length - 1;
      while ( len-- &gt; 0 ) args[ len ] = arguments[ len + 1 ];
      if (shouldNotBeStubbed(el, stubs, modifiedComponents)) {
        return originalCreateElement.apply(void 0, [ el ].concat( args ))
      }

      if (isConstructor(el) || isComponentOptions(el)) {
        if (stubAllComponents) {
          var stub = createStubFromComponent(el, el.name || 'anonymous', _Vue);
          return originalCreateElement.apply(void 0, [ stub ].concat( args ))
        }
        var Constructor = shouldExtend(el, _Vue) ? extend(el, _Vue) : el;

        return originalCreateElement.apply(void 0, [ Constructor ].concat( args ))
      }

      if (typeof el === 'string') {
        var original = resolveComponent(el, originalComponents);

        if (!original) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        if (isDynamicComponent(original)) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        var stub$1 = createStubIfNeeded(stubAllComponents, original, _Vue, el);

        if (stub$1) {
          Object.assign(vm.$options.components, ( obj = {}, obj[el] = stub$1, obj));
          modifiedComponents.add(el);
        }
      }

      return originalCreateElement.apply(void 0, [ el ].concat( args ))
    }-->
`;

exports[`renders correctly 4`] = `
<div>
  <button
    class="white--text v-btn theme--light blue-grey"
    type="button"
  >
    <div
      class="v-btn__content"
    >
      <i
        aria-hidden="true"
        class="v-icon v-icon--right material-icons theme--dark"
      >
        add
      </i>
      
    Add More Currency
  
    </div>
  </button>
   
  <div
    style="display: none;"
  >
    <input
      class="input-isi"
      placeholder="Input Currency"
    />
     
    <button
      class="white--text v-btn theme--light blue-grey"
      type="button"
    >
      <div
        class="v-btn__content"
      >
        
      Submit
      
      </div>
    </button>
  </div>
</div>
`;
